common/src/main/java/org/conscrypt/NativeCrypto.java

```diff
@@ -925,55 +928,50 @@ static native void SSL_set_client_CA_list(long ssl, NativeSsl ssl_holder, byte[]
     };
 
     /** Protocols to enable by default when "TLSv1.1" is requested. */
-    static final String[] TLSV11_PROTOCOLS = new String[] {
-            SUPPORTED_PROTOCOL_TLSV1,
-            SUPPORTED_PROTOCOL_TLSV1_1,
-            SUPPORTED_PROTOCOL_TLSV1_2,
-    };
+    static final String[] TLSV11_PROTOCOLS = TLSV12_PROTOCOLS;
 
     /** Protocols to enable by default when "TLSv1" is requested. */
-    static final String[] TLSV1_PROTOCOLS = new String[] {
-            SUPPORTED_PROTOCOL_TLSV1,
-            SUPPORTED_PROTOCOL_TLSV1_1,
-            SUPPORTED_PROTOCOL_TLSV1_2,
-    };
+    static final String[] TLSV1_PROTOCOLS = TLSV11_PROTOCOLS;
 
     static final String[] DEFAULT_PROTOCOLS = TLSV12_PROTOCOLS;
+    private static final String[] SUPPORTED_PROTOCOLS = DEFAULT_PROTOCOLS;
 
     static String[] getSupportedProtocols() {
-        return TLSV12_PROTOCOLS.clone();
+        return SUPPORTED_PROTOCOLS.clone();
     }
 
     static void setEnabledProtocols(long ssl, NativeSsl ssl_holder, String[] protocols) {
         checkEnabledProtocols(protocols);
-        // openssl uses negative logic letting you disable protocols.
-        // so first, assume we need to set all (disable all) and clear none (enable none).
-        // in the loop, selectively move bits from set to clear (from disable to enable)
-        long optionsToSet = (NativeConstants.SSL_OP_NO_SSLv3 | NativeConstants.SSL_OP_NO_TLSv1
-                | NativeConstants.SSL_OP_NO_TLSv1_1 | NativeConstants.SSL_OP_NO_TLSv1_2);
-        long optionsToClear = 0;
-        for (String protocol : protocols) {
-            if (protocol.equals(SUPPORTED_PROTOCOL_TLSV1)) {
-                optionsToSet &= ~NativeConstants.SSL_OP_NO_TLSv1;
-                optionsToClear |= NativeConstants.SSL_OP_NO_TLSv1;
-            } else if (protocol.equals(SUPPORTED_PROTOCOL_TLSV1_1)) {
-                optionsToSet &= ~NativeConstants.SSL_OP_NO_TLSv1_1;
-                optionsToClear |= NativeConstants.SSL_OP_NO_TLSv1_1;
-            } else if (protocol.equals(SUPPORTED_PROTOCOL_TLSV1_2)) {
-                optionsToSet &= ~NativeConstants.SSL_OP_NO_TLSv1_2;
-                optionsToClear |= NativeConstants.SSL_OP_NO_TLSv1_2;
-            } else if (protocol.equals(OBSOLETE_PROTOCOL_SSLV3)) {
-                // Do nothing since we no longer support this protocol, but
-                // allow it in the list of protocols so we can give an error
-                // message about it if the handshake fails.
-            } else {
-                // error checked by checkEnabledProtocols
-                throw new IllegalStateException();
+        List<String> protocolsList = Arrays.asList(protocols);
```

Comment:
Sanity check: Are there existing tests which cover this function? It contains some non-trivial mapping of protocol lists to version ranges. Stuff like what does it mean to enable 1.0, 1.2, but not 1.1.

Reply:
Yep, there are tests in SSLSocketTest and SSLEngineTest that cover protocol negotiation, including noncontiguous sets of protocols.